src/dune_engine/build_system.ml

```diff
@@ -2256,6 +2263,18 @@ end = struct
       ~input:(module Rule)
       (execute_rule_impl ~rule_kind:Normal_rule)
 
+  let rule_source () =
```

Comment:
This is very similar to `Rule_fn.loc`. I don't think we need both functions. We should be able to unify them.